pkg/webui/account/containers/profile-settings-form/index.js

```diff
@@ -0,0 +1,312 @@
+        '_profile_picture_source',
+      ])
+      if (Object.keys(patch).length === 0) {
+        patch = { ...values }
+        delete patch.profile_picture
+        delete patch._profile_picture_source
+      }
+      try {
+        const updatedUser = await dispatch(
+          attachPromise(updateUser({ id: user.ids.user_id, patch })),
+        )
+        resetForm({
+          values: {
+            ...user,
+            ...updatedUser,
+            _profile_picture_source: values._profile_picture_source,
+          },
+        })
+        toast({
+          title: sharedMessages.success,
+          message: m.successMessage,
+          type: toast.types.SUCCESS,
+        })
+      } catch (error) {
+        setError(error)
+        setSubmitting(false)
+      }
+    },
+    [dispatch, user, validationContext],
+  )
+
+  const handleDelete = useCallback(async () => {
+    try {
+      await dispatch(promisifiedDeleteUser())
+
+      // The hard redirect will conclude the deletion by deleting the
+      // (now invalid) session cookie and redirecting back to the login screen.
+      // The `account-deleted` query will cause a success notification to be
+      // shown on the login screen.
+      const appRoot = selectApplicationRootPath()
+      window.location = `${appRoot}/login?account-deleted`
+    } catch {
+      toast({
+        title: m.deleteAccount,
+        message: m.deleteAccountError,
+        type: toast.types.ERROR,
+      })
+    }
+  }, [dispatch])
+
+  const initialValues = validationSchema.cast(
+    { profile_picture: undefined, ...user },
+    { context: validationContext },
+  )
+
+  return (
+    <Form
+      initialValues={initialValues}
+      validationSchema={validationSchema}
+      validationContext={validationContext}
+      onSubmit={handleSubmit}
+      error={error}
+      validateOnChange
+    >
+      {useGravatarConfig && !disableUploadConfig && (
+        <Form.Field
+          name="_profile_picture_source"
+          title={m.profilePicture}
+          component={Radio.Group}
+          onChange={setProfilePictureSource}
+        >
+          <Radio label={m.useGravatar} value="gravatar" />
+          <Radio label={m.uploadAnImage} value="upload" />
+        </Form.Field>
+      )}
+      {!disableUploadConfig && profilePictureSource === 'upload' && (
+        <Form.Field
+          name="profile_picture"
+          component={FileInput}
+          title={m.imageUpload}
+          message={m.chooseImage}
+          changeMessage={m.changeImage}
+          providedMessage={m.imageProvided}
+          accept={['.jpg', '.jpeg', '.png']}
+          decode={imageDataDecoder}
+          dataTransform={imageDataTransform}
+          imageClassName={style.uploadImagePreview}
+          required={useGravatarConfig}
+          mayRemove
+          image
+        />
+      )}
+      {useGravatarConfig && profilePictureSource === 'gravatar' && (
```

Comment:
When `use_gravatar` and `disable_upload` are `true` this part is rendered and this is shown to the user:

<img width="525" alt="Screenshot 2021-02-08 at 16 37 45" src="https://user-images.githubusercontent.com/16374166/107237300-41825600-6a2f-11eb-8394-041dfb4837e3.png">

> You can upload a different profile picture by selecting the option above.

There is no option above to upload profile picture.